### PR TITLE
Allow HTTP Basic Auth header on CONNECT

### PR DESCRIPTION
**Pull request by shanecurran**

### CODE REVIEW

File: src/proxy.rs [hunk 1]
```diff
@@ -60,10 +62,21 @@ macro_rules! make_service {
             let additional_host_mapping = additional_host_mapping.clone();
             let additional_root_certificates = additional_root_certificates.clone();
             let client_addr = conn.remote_addr();
-
             async move {
                 Ok::<_, Error>(service_fn(move |mut req: Request<Body>| {
                     log::info!("Received request to connect: {}", req.uri());
+                    let auth = match req.headers().get(header::PROXY_AUTHORIZATION) {
+                        Some(header) => {
+                            let header_str = header.clone();
```
Comment (shanecurran) [inline src/proxy.rs]:
```suggestion
```

Comment (shanecurran) [inline src/proxy.rs]:
```suggestion
```

File: src/proxy.rs [hunk 2]
```diff
@@ -60,10 +62,20 @@ macro_rules! make_service {
...
             async move {
                 Ok::<_, Error>(service_fn(move |mut req: Request<Body>| {
                     log::info!("Received request to connect: {}", req.uri());
+                    let auth = match req.headers().get(header::PROXY_AUTHORIZATION) {
+                        Some(header) => {
+                            let header_val = match header.to_str() {
+                                Ok(s) => s.to_string(),
+                                Err(_) => "".to_string(),
+                            };
+                            Some(header_val)
```
Comment (lfarrel6) [inline src/proxy.rs]:
```suggestion
                            header.to_str().ok().map(|header_str| header_str.to_string())
```

**Review by lfarrel6 — CHANGES_REQUESTED** [review]
1 suggestion

**Review by lfarrel6 — APPROVED** [review]

**Pull request merged**